gradle-versions-plugin/src/main/kotlin/com/github/benmanes/gradle/versions/updates/Resolver.kt

```diff
@@ -402,10 +401,10 @@ class Resolver(
 
   private fun getResolvableDependencies(configuration: Configuration): List<Coordinate> {
     val coordinates = configuration.dependencies
-      .filter { dependency -> dependency is ExternalDependency }
-      .map { dependency ->
+      .filterIsInstance<ExternalDependency>()
+      .mapTo(mutableListOf()) { dependency ->
         Coordinate.from(dependency)
-      } as MutableList<Coordinate>
```

Comment:
I would guess that this uses a different factory method now, causing the constraint reason to be lost.

Reply:
I believe it was something different, see the update.